pkg/controller/ec2/transitgateway/setup.go

```diff
@@ -138,19 +139,32 @@ func (t *tagger) Initialize(ctx context.Context, mgd resource.Managed) error {
 	if !ok {
 		return errors.New(errUnexpectedObject)
 	}
-	added := false
+	var transitGatewayTags svcapitypes.TagSpecification
+	for _, tagSpecification := range cr.Spec.ForProvider.TagSpecifications {
+		if aws.StringValue(tagSpecification.ResourceType) == "transit-gateway" {
+			transitGatewayTags = *tagSpecification
+		}
+	}
+
 	tagMap := map[string]string{}
-	for _, t := range cr.Spec.ForProvider.Tags {
-		tagMap[awsclients.StringValue(t.Key)] = awsclients.StringValue(t.Value)
+	tagMap["Name"] = cr.Name
+	for _, t := range transitGatewayTags.Tags {
+		tagMap[aws.StringValue(t.Key)] = aws.StringValue(t.Value)
 	}
 	for k, v := range resource.GetExternalTags(mgd) {
-		if tagMap[k] != v {
-			cr.Spec.ForProvider.Tags = append(cr.Spec.ForProvider.Tags, svcapitypes.Tag{Key: awsclients.String(k), Value: awsclients.String(v)})
-			added = true
-		}
+		tagMap[k] = v
 	}
-	if !added {
-		return nil
+	transitGatewayTags.Tags = make([]*svcapitypes.Tag, len(tagMap))
+	transitGatewayTags.ResourceType = aws.String("transit-gateway")
+	i := 0
+	for k, v := range tagMap {
+		transitGatewayTags.Tags[i] = &svcapitypes.Tag{Key: aws.String(k), Value: aws.String(v)}
+		i++
 	}
+	sort.Slice(transitGatewayTags.Tags, func(i, j int) bool {
```

Comment:
Sorting messes up with composition patches where users patch the given index. I'd suggest appending the tag if it's not found, similar to this implementation https://github.com/crossplane/provider-aws/blob/master/pkg/controller/ecr/repository/controller.go#L258

Reply:
I will check